Utilities/WDLs/DownsampleAndCollectCoverage.wdl

```diff
@@ -0,0 +1,172 @@
+                ref_fasta_index = ref_fasta_index,
+                read_length = read_length,
+                use_fast_algorithm = use_fast_algorithm,
+                docker = docker,
+                picard_jar_override = picard_jar_override,
+                preemptible = preemptible
+        }
+    }
+
+    call Downsample {
+        input:
+            input_cram = input_cram,
+            downsample_probability = downsample_probability,
+            target_coverage = target_coverage,
+            original_coverage = CollectOriginalCoverage.mean_coverage,
+            downsample_strategy = downsample_strategy,
+            ref_fasta = ref_fasta,
+            ref_fasta_index = ref_fasta_index,
+            docker = docker,
+            picard_jar_override = picard_jar_override,
+            preemptible = preemptible
+    }
+
+    call CollectWgsMetrics as CollectDownsampledCoverage {
+        input:
+            input_cram = Downsample.downsampled_cram,
+            input_cram_index = Downsample.downsampled_cram_index,
+            coverage_intervals = coverage_intervals,
+            ref_fasta = ref_fasta,
+            ref_fasta_index = ref_fasta_index,
+            read_length = read_length,
+            use_fast_algorithm = use_fast_algorithm,
+            docker = docker,
+            picard_jar_override = picard_jar_override,
+            preemptible = preemptible
+    }
+
+    output {
+        File downsampled_cram = Downsample.downsampled_cram
+        File downsampled_cram_index = Downsample.downsampled_cram_index
+        Float downsampled_mean_coverage = CollectDownsampledCoverage.mean_coverage
+        Float? original_mean_coverage = CollectOriginalCoverage.mean_coverage
+    }
+}
+
+task CollectWgsMetrics {
+    input {
+        File input_cram
+        File input_cram_index
+        File? coverage_intervals
+        File ref_fasta
+        File ref_fasta_index
+        Int read_length
+        Boolean use_fast_algorithm
+
+        String docker
+        File? picard_jar_override
+
+        Int preemptible
+    }
+
+    Int disk_size = ceil(size(input_cram, "GiB") + size(ref_fasta, "GiB") + size(ref_fasta_index, "GiB")) + 50
+
+    String output_basename = sub(sub(basename(input_cram), "\\.bam$", ""), "\\.cram$", "")
+
+    command <<<
+        set -e -o pipefail
+
+        ~{if defined(picard_jar_override) then "java -Xms2000m -Xmx2500m -jar " + picard_jar_override else 'gatk --java-options "-Xms2000m -Xmx2500m"'} \
+        CollectWgsMetrics \
+        -INPUT ~{input_cram} \
+        -VALIDATION_STRINGENCY SILENT \
+        -REFERENCE_SEQUENCE ~{ref_fasta} \
+        ~{"-INTERVALS " + coverage_intervals} \
+        -OUTPUT ~{output_basename}.wgs_metrics \
+        -READ_LENGTH ~{read_length} \
+        -USE_FAST_ALGORITHM ~{use_fast_algorithm}
+
+        cat ~{output_basename}.wgs_metrics | grep -v -e "^#" -e "^$" -e "^GENOME_TERRITORY" | head -n 1 | awk '{ print $2 }' > ~{output_basename}.mean_coverage
```

Comment:
I can see this works though seeing the constant string `GENOME_TERRITORY` feels a bit messy. I don't think it's necessary. And the `$2` is technically not robust against changes in Picard's output format (as if that would ever happen), but at least this should still work:

```
grep -v -e '^#' -e "^$" "~{output_basename}.wgs_metrics" | head -2 | awk '{ print $2 }' | tail -1
```
If you really wanted to avoid the hardcoded `$2` in the awk script, you could do something like:
```
grep -v -e '^#' -e "^$" "~{output_basename}.wgs_metrics" | head -2 > wgs.tsv
COL_NUM=$(sed 's/\t/\n/g' | grep -n 'MEAN_COVERAGE' | cut -d':' -f1)
awk -v col=$COL_NUM ' { print $col }' wgs.tsv | tail -1 > "~{output_basename}.mean_coverage"
```
but I think this is probably more trouble than it's worth.